### PR TITLE
Put file split

### PR DESCRIPTION
**Pull request by jdoliner**

### CODE REVIEW

File: src/client/pfs/pfs.proto [hunk 1]
```diff
@@ -215,6 +215,16 @@ message PutFileRequest {
   string url = 5;
   // applies only to URLs that can be recursively walked, for example s3:// URLs
   bool recursive = 6;
+  // Delimiter causes data to be broken up into separate files with File.Path
+  // as a prefix.
+  Delimiter delimiter = 7;
+  // TargetChunkCount specifies the target number of datums in each written
```
Comment (derekchiang) [inline src/client/pfs/pfs.proto]: Wrong name

File: src/client/pfs/pfs.proto [hunk 2]
```diff
@@ -215,6 +215,16 @@ message PutFileRequest {
...
+  // file it may be lower if data does not split evenly, but will never be
+  // higher, unless the value is 0.
+  int64 target_file_datums = 8;
+  // TargetChunkBytes specifies the target number of bytes in each written
```
Comment (derekchiang) [inline src/client/pfs/pfs.proto]: Same here

File: src/server/pfs/fuse/filesystem.go [hunk 3]
```diff
@@ -333,11 +333,13 @@ func (f *file) delimiter() pfsclient.Delimiter {
 }
 
 func (f *file) touch() error {
```
Comment (derekchiang) [inline src/server/pfs/fuse/filesystem.go]: Does this mean that if you save a file as `foo.json` in a FUSE mount, we somehow end up with a directory `foo`?

File: src/server/pachyderm_test.go [hunk 4]
```diff
@@ -786,6 +786,97 @@ func TestPipelineJobDeletion(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestPutFileSplit(t *testing.T) {
```
Comment (derekchiang) [inline src/server/pachyderm_test.go]: Shouldn't we also check that the children files are named the way they are supposed to?  Right now looks like we are only checking the number of children.

Comment (derekchiang) [inline src/server/pachyderm_test.go]: Also this should be put into `server_test.go` which is for PFS-only tests.

File: src/server/pfs/fuse/filesystem.go (hunk 3 above)
Comment (jdoliner) [inline src/server/pfs/fuse/filesystem.go]: Yes, which is arguably pretty weird. I think we'll just not use split through fuse.

**Comment by jdoliner** [comment]
Alright all comments are addressed.

**Pull request merged**